sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -30,34 +30,33 @@
 class KatibClient(object):
     def __init__(
         self,
-        config_file=None,
-        context=None,
-        client_configuration=None,
-        persist_config=True,
+        config_file: str = None,
+        context: str = None,
+        client_configuration: client.Configuration = None,
     ):
         """KatibClient constructor.
 
         Args:
-            config_file: Name of the kube-config file. Defaults to ~/.kube/config.
+            config_file: Path to the kube-config file. Defaults to ~/.kube/config.
             context: Set the active context. Defaults to current_context from the kube-config.
-            client_configuration: The kubernetes.client.Configuration to set configs to.
-            persist_config: If True, config file will be updated when changed.
+            client_configuration: Client configuration for cluster authentication.
+                You have to provide valid configuration with Bearer token or
+                with username and password.
+                You can find an example here: https://github.com/kubernetes-client/python/blob/67f9c7a97081b4526470cad53576bc3b71fa6fcc/examples/remote_cluster.py#L31
         """
 
-        self.in_cluster = None
-        if config_file or not utils.is_running_in_k8s():
-            config.load_kube_config(
-                config_file=config_file,
-                context=context,
-                client_configuration=client_configuration,
-                persist_config=persist_config,
-            )
-            self.in_cluster = False
-        else:
-            config.load_incluster_config()
-            self.in_cluster = True
+        self.in_cluster = False
```

Comment:
Do we need a var, `in_cluster`?

Reply:
@tenzen-y Currently, we use it when Katib is deployed as part of Kubeflow Central Dashboard: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L112-L124.
Although, it can generate incorrect links when user relies on `load_incluster_config()` and not run it from Kubeflow Notebooks.
We will discuss about it in the future.

Reply:
I see.
Thanks for clarifying! Could you create an issue to keep tracking this?

Reply:
Sure, let's track this.